bootstrap/v2/pkg/kfapp/existing_arrikto/existing.go

```diff
@@ -470,3 +480,24 @@ func genRandomString(length int) string {
 	}
 	return string(b)
 }
+
+func initialiseHMACSecretFromEnvOrGen(secEnv string, reqLen int) []byte {
+	envContent := os.Getenv(secEnv)
+
+	if len(envContent) < reqLen {
+		log.Println("WARNING: HMAC secret not provided or secret too short. Generating a random one from nonce characters.")
+		return []byte(createNonce(reqLen))
+	}
+
+	return []byte(envContent)
+}
+
+func createNonce(length int) string {
```

Comment:
You can use the genRandomString function here and the createNonce is not really required.